contrib/jenkins_tests/globals.sh

```diff
@@ -40,7 +41,7 @@ style_dir=${WORKSPACE}/${prefix}/style
 nproc=$(grep processor /proc/cpuinfo|wc -l)
 make_opt="-j$(($nproc / 2 + 1))"
 if [ $(command -v timeout >/dev/null 2>&1 && echo $?) ]; then
-    timeout_exe="timeout -s SIGKILL 20m"
+    timeout_exe="timeout -s SIGKILL 40m"
```

Comment:
please do 20m and use `timeout -s SIGKILL 40m` directly in places with long duration.

Reply:
Moved long timeout to `do_check_result` function that is used only for compiler invocation.